### ISSUE TITLE
Raw identifier keywords

### ISSUE THREAD

**Issue by michaelfairley** [serde-rs/serde#1362]
Right now, `serde_derive` passes through the `r#` on [raw identifiers](https://rust-lang-nursery.github.io/edition-guide/rust-2018/module-system/raw-identifiers.html) when the identifier is a keyword.

For example ([playground link](https://play.rust-lang.org/?gist=0492ec75839fb4634b7a9ee6e9a1094f&version=nightly&mode=debug&edition=2018))
```rust
#[derive(Serialize)]
struct Foo {
    r#type: i32,
    r#bar: i32
}

fn main() {
    let foo = Foo{ r#type: 3, r#bar: 12 };
    println!("{}", serde_json::to_string_pretty(&foo).unwrap());
}
```

prints
```
{
  "r#type": 3,
  "bar": 12
}
```

That `r#` on the `r#type` seems like it shouldn't be exposed downstream of serde, since the current-list-of-keywords-in-Rust is an implementation detail that serialization probably shouldn't care about. Currently, a bunch of the data format libraries that work with serde (e.g. [ron](https://github.com/ron-rs/ron)) emit invalid data when provided with these raw identifiers, since they assume that field identifiers have a limited set of characters (that doesn't include `#`).

Unfortunately, the root of this issue is probably upstream of serde, and even writing a test case for this in this repo is tricky (since the test code has to be Rust 2018).

**Comment by dtolnay** [comment]
Interesting! This is a bug. That program should be printing `"type": 3` instead of `"r#type": 3`.

Would you be interested in tracking down where this is going wrong and sending a PR? Otherwise I should be able to take a look later this week.

As a workaround, at least `serde(rename = "...")` seems to work but this shouldn't be necessary.

```rust
#[derive(Serialize)]
struct Foo {
    #[serde(rename = "type")]
    r#type: i32,
    r#bar: i32
}
```

**Comment by dtolnay** [comment]
I published 1.0.73 with this fix. Thanks! The code in the playground link will print:

```json
{
  "type": 3,
  "bar": 12
}
```
